peachjam/models/user_following.py

```diff
@@ -195,116 +198,53 @@ def get_documents_queryset(self):
             topics = [self.taxonomy] + [t for t in self.taxonomy.get_descendants()]
             return qs.filter(taxonomies__topic__in=topics)
 
-    def get_new_followed_documents(self):
+    def get_new_followed_documents(self, since=None, limit=10):
         qs = self.get_documents_queryset().preferred_language(
             self.user.userprofile.preferred_language.iso_639_3
         )
-        if self.last_alerted_at:
+        cutoff = since or self.last_alerted_at
+        if cutoff:
             qs = qs.filter(created_at__gt=self.last_alerted_at)
+        return qs[:limit]
+
+    def create_timeline_event(self, documents):
+        if not documents:
+            log.info("No documents")
+            return
+        log.info(f"Found {documents.count()} new documents for {self.followed_object}")
+        # check for unsent event
+        event = TimelineEvent.objects.filter(
+            user_following=self,
+            event_type=TimelineEvent.EventTypes.NEW_DOCUMENTS,
+            email_alert_sent_at__isnull=True,
+        ).first()
+        if not event:
+            log.info(f"Creating new timeline event for {self.followed_object}")
+            event = TimelineEvent.objects.create(
+                user_following=self,
+                event_type=TimelineEvent.EventTypes.NEW_DOCUMENTS,
+            )
+        else:
+            log.info(f"Updating existing timeline event for {self.followed_object}")
+
+        event.subject_documents.add(*documents)
+        self.last_alerted_at = timezone.now()
+        self.save(update_fields=["last_alerted_at"])
```

Comment:
this isn't a good place to do this -- it's not related to creating the timeline events. Whatever method is responsible for triggering this update should, once the event is created, then update the `last_alerted_at`.